tests/11register.pl

```diff
@@ -282,3 +282,49 @@
          Future->done( 1 );
       });
    };
+
+test "registration accepts non-ascii passwords",
+   requires => [ $main::API_CLIENTS[0], localpart_fixture() ],
+
+   do => sub {
+      my ( $http, $localpart ) = @_;
+
+      my $session;
+
+      $http->do_request_json(
+         method => "POST",
+         uri    => "/r0/register",
+
+         content => {
+            username => $localpart,
+            password => "übers3kr1t",
+            device_id => "xyzzy",
+            initial_device_display_name => "display_name",
+         },
+      )->main::expect_http_401->then( sub {
+         my ( $response ) = @_;
+
+         my $body = decode_json $response->content;
+
+         assert_json_keys( $body, qw( session ));
+
+         $session = $body->{session};
```

Comment:
`$session` doesn't seem to be used outside this block so you might as well just declare it here; `my $session`.